tony-portal/app/history/HistoryFileMover.java

```diff
@@ -99,4 +151,65 @@ private void moveIntermediateToFinished(FileSystem fs, FileStatus[] jobDirs, Zon
   private boolean jobInProgress(String jhistFileName) {
     return !jhistFileName.endsWith(Constants.HISTFILE_SUFFIX);
   }
+
+  // This function queries and returns Yarn for the App IDs of apps with YarnApplicationState.KILLED
+
+  private List<String> getKilledApps() throws IOException, YarnException {
+    List<String> totalKilledAppIds = new ArrayList<String>();
+    long currentTimestamp = System.currentTimeMillis();
+
+    //set timeDelta for 36 hours
+    long timeDelta = 36 * 60 * 60 * 1000;
+    YarnConfiguration yarnConf = new YarnConfiguration();
+    if (System.getenv(Constants.HADOOP_CONF_DIR) != null) {
+      yarnConf.addResource(new Path(System.getenv(Constants.HADOOP_CONF_DIR) +
+          File.separatorChar + Constants.CORE_SITE_CONF));
+      yarnConf.addResource(new Path(System.getenv(Constants.HADOOP_CONF_DIR) +
+          File.separatorChar + Constants.YARN_SITE_CONF));
+    }
+    YarnClient yarnClient = YarnClient.createYarnClient();
+    yarnClient.init(yarnConf);
+    yarnClient.start();
+    List<ApplicationReport> totalKilledApps = yarnClient.getApplications(EnumSet.of(YarnApplicationState.KILLED));
+
+    // Get apps within a specified time period. This prevents the History File mover from iterating through
```

Comment:
I'm not sure this logic is required. In fact if an app runs for > 36 hrs, then is killed, it will be missed in this filter, right?

Also, there's the case where an app is killed from RM point of view, but it's purged from RM memory. Then when we query yarnClient.getApplications(KILLED), it will not be included in this query, and this app will also be missed.

How about this alternative: query for apps which are not completed (in this case, NEW, NEW_SAVING, SUBMITTED, ACCEPTED, or RUNNING). These will definitely be returned by RM when queried. Then add everything else to the set of apps to be marked KILLED (and moved from intermediate dir to done dir).

(This assumes that all apps which were not killed, but finished properly (i.e. FAILED or FINISHED) have already been moved out of the intermediate dir.)

I think this second concern can be addressed in a different PR...but I think we can remove the 36 hour filter.

Reply:
@hungj 
-removed the 36hr logic. 
-added query for apps that are killed by user or RM
-updated HistoryFileMover to include both killed apps check and move intermediate files to finished in the same scheduledThreadPool